include/zephyr/devicetree.h

```diff
@@ -858,44 +858,77 @@
  * Example usage:
  *
  * @code{.c}
- *     DT_ENUM_IDX(DT_NODELABEL(usb1), maximum_speed) // 1
- *     DT_ENUM_IDX(DT_NODELABEL(usb2), maximum_speed) // 3
+ *     DT_ENUM_IDX_BY_IDX(DT_NODELABEL(usb1), maximum_speed, 0) // 1
+ *     DT_ENUM_IDX_BY_IDX(DT_NODELABEL(usb2), maximum_speed, 0) // 3
```

Comment:
(optional) nit - could you given an example of a nonzero index ideally (even more optional) including a snippet of DT that corresponds to the sample code? Similarly for the other macros, but won't mention it.

Reply:
Well, I changed the current one...